yktrace/src/hwt/mapper.rs

```diff
@@ -159,6 +154,9 @@ impl HWTMapper {
 
         ents.sort_by(|x, y| x.range.start.partial_cmp(&y.range.start).unwrap());
         for ent in ents {
+            if ent.value.bb == NO_BASICBLOCK {
```

Comment:
I'm not 100% sure.

The facts:

 * Lukas and I hope and think this can't happen, but we are not certain.
 * If it can happen, and we don't have this check in release code, we may emit and execute semantically incorrect traces.

Thoughts?

Reply:
We never compile `assert`s out, even in release mode.

Reply:
Indeed. So you'd prefer a regular assert?

Reply:
Yes but only if you're *super super* unsure which, in this case, I think might actually be reasonable. In nearly all other cases, `debug_assert` is the right choice IMHO.

Reply:
Agreed on a regular `assert` for the time being, until we know more about when and how this can happen.

Reply:
763caac5b054 should be a decent compromise.